cli/cmd/component_dev.go

```diff
@@ -0,0 +1,359 @@
+
+func runComponentsDevMode(_ *cobra.Command, args []string) error {
+	cli.StartProgress("Loading components state...")
+	var err error
+	cli.LwComponents, err = lwcomponent.LoadState(cli.LwApi)
+	cli.StopProgress()
+	if err != nil {
+		return errors.Wrap(err, "unable to load components")
+	}
+
+	component, found := cli.LwComponents.GetComponent(args[0])
+	if !found {
+		component = &lwcomponent.Component{
+			Name: args[0],
+		}
+
+		if component.UnderDevelopment() {
+			return errors.New("component already under development.")
+		}
+
+		cli.OutputHuman("Component '%s' not found. Defining a new component.\n",
+			color.HiYellowString(component.Name))
+
+		var (
+			helpMsg = fmt.Sprintf("What are these component types ?\n"+
+				"\n'%s' - A binary accessible via the Lacework CLI (Users will run 'lacework <COMPONENT_NAME>')"+
+				"\n'%s' - A regular standalone-binary (this component type is not accessible via the CLI)"+
+				"\n'%s' - A library that only provides content for the Lacework CLI or other components\n",
+				lwcomponent.CommandType, lwcomponent.BinaryType, lwcomponent.LibraryType)
+		)
+		if cdkDevState.Type == "" {
+			if err := survey.AskOne(&survey.Select{
+				Message: "Select the type of component you are developing:",
+				Help:    helpMsg,
+				Options: []string{
+					lwcomponent.CommandType,
+					lwcomponent.BinaryType,
+					lwcomponent.LibraryType,
+				},
+			}, &cdkDevState.Type); err != nil {
+				return err
+			}
+		}
+
+		component.Type = lwcomponent.Type(cdkDevState.Type)
+
+		if cdkDevState.Description == "" {
+			if err := survey.AskOne(&survey.Input{
+				Message: "What is this component about? (component description):",
+			}, &component.Description); err != nil {
+				return err
+			}
+		} else {
+			component.Description = cdkDevState.Description
+		}
+	}
+
+	if err := component.EnterDevelopmentMode(); err != nil {
+		return errors.Wrap(err, "unable to enter development mode")
+	}
+
+	rPath, err := component.RootPath()
+	if err != nil {
+		return errors.New("unable to detect RootPath")
+	}
+
+	cli.OutputHuman("Component '%s' in now in development mode.\n",
+		color.HiYellowString(component.Name))
+
+	if component.Type == lwcomponent.CommandType {
+		// Offer the creation of a component scaffolding
+		if cdkDevState.Scaffolding == "" && cli.InteractiveMode() {
+			if err := survey.AskOne(&survey.Select{
+				Message: "Would you like to initialize your component with a scaffolding? ",
+				Options: []string{"No. Start from scratch", "Golang", "Python"},
+			}, &cdkDevState.Scaffolding); err != nil {
+				return err
+			}
+		}
+
+		switch cdkDevState.Scaffolding {
+		case "Golang":
+			if err := cdkGolangScaffolding(component); err != nil {
+				return err
+			}
+
+		case "Python":
+			// TBA
+			cli.OutputHuman("\n (:sadpanda:) We are still working on a Python Scaffolding.")
+			cli.OutputHuman("\n\nIn the meantime use our Python SDK: %s\n",
+				color.HiMagentaString("https://github.com/lacework/python-sdk"))
```

Comment:
Would it be useful to have a named func for outputting hyperlinks?

Reply:
Could you give me an example?